src/observe-tag.js

```diff
@@ -109,17 +112,25 @@ module.exports = tagFunction => {
 					}
 
 					// also set the scrollLeft and scrollTop (since this is reset to 0 by default)
-					elementToGiveFocus.scrollLeft = removedElementWithFocusData.scrollLeft
-					elementToGiveFocus.scrollTop = removedElementWithFocusData.scrollTop
+					if (removedElementWithFocusData.scrollLeft || removedElementWithFocusData.scrollTop) {
+						elementToGiveFocus.scrollLeft = removedElementWithFocusData.scrollLeft
+						elementToGiveFocus.scrollTop = removedElementWithFocusData.scrollTop
+					}
 				}
 			}
 
 			// copy the reaction and effects from the old tag to the new one
 			tagResult[TRAM_TAG_REACTION] = oldTag[TRAM_TAG_REACTION]
 			tagResult[TRAM_TAG_NEW_EFFECTS] = oldTag[TRAM_TAG_NEW_EFFECTS]
 			tagResult[TRAM_TAG_CLEANUP_EFFECTS] = oldTag[TRAM_TAG_CLEANUP_EFFECTS]
+
+			// both these actions cause forced reflow, and can be performance issues
+			oldTag.replaceWith(tagResult)
+			if (elementToGiveFocus) elementToGiveFocus.focus()
```

Comment:
this is still causing issues, and I have no idea how to performantly fix it. We need to replace before we focus, because you can't focus an element that isn't in the document. If there is a way we can batch it, that would be magic, but I haven't found anything that offers that (yet).
![image](https://user-images.githubusercontent.com/326557/128259424-0ccc32f9-7db9-4b43-8c2a-1ceda1dd6d97.png)

Reply:
I've asked this question on stack-overflow, so we'll see if we get a response - https://stackoverflow.com/questions/68675817/how-can-i-group-javascript-actions-that-force-reflow